test/functional/tests/io_class/test_io_class_occupancy_stress.py

```diff
@@ -0,0 +1,177 @@
+        description:
+          Trigger IO to cache and verify if doesn't loose ability to saturate
+          fully.
+        pass_criteria:
+          - Cache is able to saturate fully
+    """
+    with TestRun.step("Prepare disks"):
+        cache, core = prepare(cache_mode=cache_mode, cache_line_size=cache_line_size)
+        cache_size = cache.get_statistics().config_stats.cache_size
+
+    with TestRun.step("Disable udev"):
+        Udev.disable()
+
+    with TestRun.step(f"Preparing filesystem and mounting {core.system_path} at {mountpoint}"):
+        filesystem = Filesystem.xfs
+        core.create_filesystem(filesystem)
+        core.mount(mountpoint)
+        sync()
+
+    with TestRun.step("Prepare test dirs"):
+        IoclassConfig = namedtuple("IoclassConfig", "id eviction_prio max_occupancy dir_path")
+        io_classes = [
+            IoclassConfig(1, 3, 0.40, f"{mountpoint}/A"),
+            IoclassConfig(2, 4, 0.30, f"{mountpoint}/B"),
+            IoclassConfig(3, 5, 0.30, f"{mountpoint}/C"),
+        ]
+
+        for io_class in io_classes:
+            fs_utils.create_directory(io_class.dir_path, parents=True)
+
+    with TestRun.step("Remove old ioclass config"):
+        ioclass_config.remove_ioclass_config()
+        ioclass_config.create_ioclass_config(False)
+
+    with TestRun.step("Adding default ioclasses"):
+        ioclass_config.add_ioclass(*str(IoClass.default(allocation="0.00")).split(","))
+
+    with TestRun.step("Adding ioclasses for all dirs"):
+        for io_class in io_classes:
+            ioclass_config.add_ioclass(
+                io_class.id,
+                f"directory:{io_class.dir_path}&done",
+                io_class.eviction_prio,
+                f"{io_class.max_occupancy:0.2f}",
+            )
+
+        casadm.load_io_classes(cache_id=cache.cache_id, file=ioclass_config_path)
+
+    with TestRun.step("Resetting cache stats"):
+        cache.purge_cache()
+        cache.reset_counters()
+
+    with TestRun.step(f"Create one file for each ioclass"):
+        for io_class in io_classes[0:3]:
+            run_io_dir(
+                io_class.dir_path, int((io_class.max_occupancy * cache_size) / Unit.Blocks4096)
+            )
+
+    with TestRun.step("Check if occupancy threshold is reached"):
+        for io_class in io_classes[0:3]:
+            actuall_occupancy = get_io_class_occupancy(cache, io_class.id)
+
+            occupancy_limit = (
+                (io_class.max_occupancy * cache_size)
+                .align_down(Unit.Blocks4096.get_value())
+                .set_unit(Unit.Blocks4096)
+            )
+
+            if not isclose(actuall_occupancy.value, occupancy_limit.value, rel_tol=0.1):
+                TestRun.LOGGER.error(
+                    f"Occupancy for ioclass {io_class.id} does not match. "
+                    f"Limit: {occupancy_limit}, actuall: {actuall_occupancy}"
+                )
+
+    for i in range(0, 6):
```

Comment:
You can use "for _ in TestRun.iteration(range(0, 6))" so the log will be more readable